cmd/argocd/commands/tree.go

```diff
@@ -0,0 +1,75 @@
+package commands
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"github.com/fatih/color"
+	"github.com/gosuri/uitable"
+)
+
+const (
+	firstElemPrefix = `├─`
+	lastElemPrefix  = `└─`
+	indent          = "  "
+	pipe            = `│ `
+)
+
+var (
+	gray = color.New(color.FgHiBlack)
+)
+
+func treeViewInnerGet(prefix string, tbl *uitable.Table, objs map[string]v1alpha1.ResourceNode, obj map[string][]string, parent v1alpha1.ResourceNode, mapNodeNameToResourceState map[string]*resourceState) {
+	if mapNodeNameToResourceState[parent.Kind+"/"+parent.Name] != nil {
+		value := mapNodeNameToResourceState[parent.Kind+"/"+parent.Name]
+
+		tbl.AddRow(value.Group, value.Namespace, value.Kind, fmt.Sprintf("%s%s/%s",
+			gray.Sprint(printPrefix(prefix)),
+			parent.Kind,
+			color.New(color.Bold).Sprint(parent.Name)),
+			value.Status,
+			value.Health,
+			value.Hook,
+			value.Message)
+	} else {
+		tbl.AddRow(parent.Group, parent.Namespace, parent.Kind, fmt.Sprintf("%s%s/%s",
+			gray.Sprint(printPrefix(prefix)),
+			parent.Kind,
+			color.New(color.Bold).Sprint(parent.Name)),
+			"",
+			"",
+			"",
+			"")
+
+	}
+
+	chs := obj[parent.UID]
+	for i, child := range chs {
+		var p string
+		switch i {
+		case len(chs) - 1:
+			p = prefix + lastElemPrefix
+		default:
+			p = prefix + firstElemPrefix
+		}
+		treeViewInnerGet(p, tbl, objs, obj, objs[child], mapNodeNameToResourceState)
+	}
+
+}
+
+func printPrefix(p string) string {
+	// this part is hacky af
```

Comment:
What is hacky about that line?

Reply:
It might not be the most effecient implementation.